app/models/ems_refresh/save_inventory.rb

```diff
@@ -1,5 +1,5 @@
 module EmsRefresh::SaveInventory
-  def save_ems_inventory(ems, hashes, target = nil)
+  def save_ems_inventory(ems, hashes, target = nil, mode = :refresh)
```

Comment:
how about a new method e.g. `save_ems_inventory_no_disconnect` which just calls `save_ems_inventory` with a mode other than `:refresh`. This would save you from a 4-ary method signature like
```ruby
save_ems_inventory(ems, target_hash, nil, :refresh_new_target)
```